api/v1alpha2/authorizers.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha2
+
+type Authorizer struct {
+	// A unique name identifying the extension authorization provider.
```

Comment:
As far as I understand we don't have this validation any longer, because it was done in the external authorizer reconciler that was removed. An alternative would be to look into using CEL and CRD validation.

Reply:
I think there is no way to do this validation in CEL, so we need to add it programmatically.

Reply:
I will just re-add https://github.com/kyma-project/istio/blob/6c9b9fc97166832fbf8cc031d005f02df29030f5/internal/reconciliations/external_authorizer/external_authorizer.go#L40-L48

Reply:
Hmm, maybe this kind of validation would be good to be kept inside the Aapi package, to not spread validations outside of the api package. So we would have CEL validation as our go-to solution and programmatic validation inside of the api package as well. 
Does that make sense to you?

Reply:
For now I created a separate package in internal, I don't think it should be accessible from outside of api-gateway. WDYT?

Reply:
I agree, that's a good point. 👍

Reply:
OK will add units as well